pkg/api/api.go

```diff
@@ -228,3 +229,133 @@ type Message struct {
 	MessageID ulid.ULID       `json:"message_id"`
 	Payload   json.RawMessage `json:"payload,omitempty"`
 }
+
+// Validator types can be checked for having a valid state.
+type Validator interface {
+	// Validate() checks whether this type is in a valid state.
+	Validate() error
+}
+
+// checks every field of the state, returning an error when any invalid field is encountered
+func (state State) Validate() error {
+	for _, field := range []Validator{
+		state.VisionStatus,
+		state.BallFound,
+		state.LocalizationStatus,
+		state.CPB,
+		state.Role,
+		state.RefBoxRole,
+		state.HomeGoal,
+		state.TeamColor,
+		state.CapacitorState,
+		state.Kinect1State,
+		state.Kinect2State,
+	} {
+		// the String() method is required, and also works for non-string Validators
+		// otherwise pointer comparision is used, and this doesn't work
+		if reflect.ValueOf(field).String() != reflect.Zero(reflect.TypeOf(field)).String() {
```

Comment:
There's no benefit of using reflect this way. Directly check if field is empty and validate it in the `switch`

Reply:
how am I supposed to fix this?
i cant force validator to be a string, so i cant cast to string
i cant compare Value types for whatever reason
i cant call a reflect.compare function, as it does not exist for whatever reason
nor do i see any reason to spend time on this

Reply:
In the `switch` you can just e.g. `state.TeamColor != "" && state.TeamColor.Validate() != nil:`